runner/util/result.py

```diff
@@ -15,12 +16,32 @@ def __init__(self, name: str, result: Result, reason: str):
         self.result = result
         self.reason = reason
 
+    def to_string(self):
+        return f'name: {self.name}\n\tresult: {self.result}\n\reason: {self.reason}'
+
+
+def sanitize_result_key(key: str) -> str:
+    return key.replace('/','_').replace('.','-')
+
 class AppResult:
     def __init__(self, directory: str, build: Builds, result: Result):
         self.result = result
         self.build = build
         self.directory = directory
         self.test_results = []
 
+    def set_result(self, result: Result):
+        self.result = result
+
     def add_test_results(self, test_results: List[TestResult]):
-        self.test_results = test_results
\ No newline at end of file
+        for result in test_results:
+            self.test_results.append(result)
+            if result.result is not Result.SUCCESS:
+                self.result = Result.TESTS_FAILED
+            elif self.result is not Result.SUCCESS:
+                self.result = Result.SUCCESS
```

Comment:
something's not quite right here.

Reply:
youre right, let me fix this. Need to get to the point where running tests work to catch a bunch of bugs likely but this one is very wrong